src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -225,7 +225,17 @@ class ArnoldOutput : public IECore::RefCounted
 				AiNodeSetStr( m_driver.get(), AiParamGetName( fileNameParameter ), output->getName().c_str() );
 			}
 
-			IECore::StringVectorDataPtr customAttributesData = new IECore::StringVectorData();
+			IECore::StringVectorDataPtr customAttributesData;
+			IECore::CompoundDataMap::const_iterator it = output->parameters().find( "custom_attributes" );
+			if( it == output->parameters().end() )
+			{
+				customAttributesData = new IECore::StringVectorData();
+			}
+			else
+			{
+				customAttributesData = IECore::runTimeCast<IECore::StringVectorData>( it->second );
```

Comment:
This is referencing the data within `output` directly, and then later it'll be modified directly too. The problem with this is that `output` is const and shouldn't be modified - the caller expects to get it back unchanged. It looks like we need to fix something up in Cortex so that this is caught at compile time, but in the meantime we should make sure we do the right thing. While we're at it we should probably make things more robust to the user passing non-StringVectorData "custom_attributes", since here's we'd be assigning `NULL` to `customAttributesData` and then crashing later. Something like this I think (untested) :

```
StringVectorDataPtr customAttributesData;
if( const StringVectorData *d = output->parametersData()->member<StringVectorData>( "custom_attributes" ) )
{
    customAttributesData = d->copy();
}
else
{
    customAttributesData = new StringVectorData;
}
```